src/datawave_cli/base_interactions.py

```diff
@@ -0,0 +1,40 @@
+from abc import ABC, abstractmethod
+import sys
+
+
+class BaseInteractions(ABC):
+    def __init__(self, args):
+        self.namespace = args.namespace
+        self.init_base_url(args)
+        self.init_cert(args)
+
+    def init_cert(self, args):
+        if args.key is None:
+            self.cert = args.cert
+        else:
+            self.cert = (args.cert, args.key)
+
+    def init_base_url(self, args):
+        """
+        Define the base_url for interacting with datawave accumulo based on arguments
+        passed in at call. Has the possibilities of localhost:port, IP:port, or an actual
+        url.
+        """
+        if (args.localhost):
+            self.base_url = "https://localhost:8443"
+        elif (args.ip):
+            pod_ip = self.get_pod_ip()
+            self.base_url = f"https://{pod_ip}:8443"
+        else:
+            # TODO figure out a good default for the url
+            url = f"{args.url}" if args.url != "" else None
+            if url is None:

```

Comment:
We don't need a default value here so we don't need this ternary. Additionally the conditional can just evaluate the `url` variable. It should be a string due to how the CLI grabs the arg from the env var, if you wanted to be sure you could just put it in an f string to ensure its a string. Then we can guarantee that this will either be the passed value, the environment variable, or a blank string. The truthyness of a blank string is false. So `bool(not url)` (or the below conditional) will evaluate true ONLY if they either pass an empty string for url (ie `--u ''`) or if the environment variable is not set. You might want to double check this would be have as expected but I see no reason it wouldn't work. And that's the more pythonic way to handle it.

```suggestion
            url = args.url
            if not url:

```